nexus-writer/src/hdf5_handlers/error.rs

```diff
@@ -277,14 +347,18 @@ impl From<TryFromIntError> for NexusHDF5Error {
     }
 }
 
-/// Used to allow errors which can be convertex to NexusHDF5Errors to be
-/// appended with hdf5 paths
+/// Used to allow errors which can be convertex to [NexusHDF5Error] to be
+/// appended with hdf5 paths. This is intended to be implemented by [Result<_,E>]
+/// where `E: Error + Into<NexusHDF5Error>`.
 pub(crate) trait ConvertResult<T, E>
 where
     E: Error + Into<NexusHDF5Error>,
 {
+    /// Implementation should convert the error to [NexusHDF5Error] type and add the path of `group` to it.
```

Comment:
```suggestion
    /// Converts the error to [NexusHDF5Error] type and add the path of `group` to it.
```

Document trait functions in the context of being read by the user, not the trait implementer. Otherwise you end up with two similar but different context in the way docs are written.